Sorting/intersection2.js

```diff
@@ -0,0 +1,32 @@
+//definition of intersection, built off a helper function. repetitions in subArrays won't break it.
+
+//helper function describes how to intersect two arrays.
+//checks whether every element in arr1, is in arr2 + not already in intersection array (this avoids repetitions)
+//if element satisfies this condition, it pushes the element into the output array
+function intersectTwoArrays(arr1, arr2) {
+    var result = [];
+    arr1.forEach(function(element) {
+        if (arr2.includes(element) && !result.includes(element)) {
```

Comment:
It may not seem like it, but `includes` has runtime `O(n)` because it needs to loop through the array and check whether an element exists. 

What is the runtime of `intersectTwoArrays`? (Hint: it is worse than `O(n log(n))`.)

**Bonus**: a very common interview question is to compute the intersection of two _sorted_ arrays. Sorting will typically take `O(n log(n))`. If `intersectTwoArrays` first sorts the arrays and then computes the intersection, can you see how to implement this method in time `O(n log(n))`?